src/lib/components/ui/FilterBuilder.svelte

```diff
@@ -125,8 +141,64 @@
 		}
 	}
 
+	// Centralized mapping field metadata
+	const mappingFieldConfig: Record<
+		string,
+		{
+			type: string;
+			ui_type: string;
+			options?: Array<{ value: string; label: string }>;
+		}
+	> = {
+		has_mappings: {
+			type: 'boolean',
+			ui_type: 'select',
+			options: [
+				{ value: 'true', label: 'Yes' },
+				{ value: 'false', label: 'No' }
+			]
+		},
+		mapping_count: {
+			type: 'number',
+			ui_type: 'short_text'
+		},
+		mapping_status: {
+			type: 'string',
+			ui_type: 'short_text',
+			options: [
+				{ value: 'planned', label: 'Planned' },
+				{ value: 'implemented', label: 'Implemented' },
+				{ value: 'verified', label: 'Verified' }
+			]
```

Comment:
I think it'd be good if we could pull from a shared type here across the codebase in case we change these at some point. We currently have the 3 statuses [here](https://github.com/defenseunicorns/lula/blob/197-filter-by-mapping-status/src/lib/types.ts#L37). Locally I tried something like this:
```ts
export const MAPPING_STATUS_OPTIONS = [
	{ value: 'planned', label: 'Planned' },
	{ value: 'implemented', label: 'Implemented' },
	{ value: 'verified', label: 'Verified' }
] as const;

export type MappingStatus = (typeof MAPPING_STATUS_OPTIONS)[number]['value'];
```

And then using `MAPPING_STATUS_OPTIONS` here.

Reply:
done!